src/main/java/io/nats/client/api/ConsumerLimits.java

```diff
@@ -0,0 +1,137 @@
+    }
+
+    ConsumerLimits(JsonValue vConsumerLimits) {
+        inactiveThreshold = readNanos(vConsumerLimits, INACTIVE_THRESHOLD);
+        maxAckPending = readInteger(vConsumerLimits, MAX_ACK_PENDING);
+    }
+
+    ConsumerLimits(ConsumerLimits.Builder b) {
+        this.inactiveThreshold = b.inactiveThreshold;
+        this.maxAckPending = b.maxAckPending;
+    }
+
+    /**
+     * Get the amount of time before the consumer is deemed inactive.
+     * @return the inactive threshold
+     */
+    public Duration getInactiveThreshold() {
+        return inactiveThreshold;
+    }
+
+    /**
+     * Gets the maximum ack pending configuration.
+     * @return maximum ack pending.
+     */
+    public long getMaxAckPending() {
+        return getOrUnset(maxAckPending);
+    }
+
+    public String toJson() {
+        StringBuilder sb = beginJson();
+        JsonUtils.addFieldAsNanos(sb, INACTIVE_THRESHOLD, inactiveThreshold);
+        JsonUtils.addField(sb, MAX_ACK_PENDING, maxAckPending);
+        return endJson(sb).toString();
+    }
+
+    /**
+     * Creates a builder for a placements object.
+     * @return the builder.
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Placement can be created using a Builder.
+     */
+    public static class Builder {
+        private Duration inactiveThreshold;
+        private Integer maxAckPending;
+
+        /**
+         * sets the amount of time before the consumer is deemed inactive.
+         * @param inactiveThreshold the threshold duration
+         * @return Builder
+         */
+        public Builder inactiveThreshold(Duration inactiveThreshold) {
+            this.inactiveThreshold = normalize(inactiveThreshold);
+            return this;
+        }
+
+        /**
+         * sets the amount of time before the consumer is deemed inactive.
+         * @param inactiveThreshold the threshold duration in milliseconds
+         * @return Builder
+         */
+        public Builder inactiveThreshold(long inactiveThreshold) {
+            this.inactiveThreshold = normalizeDuration(inactiveThreshold);
+            return this;
+        }
+
+        /**
+         * Sets the maximum ack pending or null to unset / clear.
+         * @param maxAckPending maximum pending acknowledgements.
+         * @return Builder
+         */
+        public Builder maxAckPending(Long maxAckPending) {
+            this.maxAckPending = normalize(maxAckPending, STANDARD_MIN);
+            return this;
+        }
+
+        /**
+         * Sets the maximum ack pending.
+         * @param maxAckPending maximum pending acknowledgements.
+         * @return Builder
+         */
+        public Builder maxAckPending(long maxAckPending) {
+            this.maxAckPending = normalize(maxAckPending, STANDARD_MIN);
+            return this;
+        }
+
+        /**
+         * Build a Placement object
+         * @return the Placement
```

Comment:
```suggestion
         * Build a ConsumerLimits object
         * @return ConsumerLimits
```

Reply:
yeah will fix, a lot of cut paste going on